lib/data/services/user_service.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_target/data/utils/net_constants.dart';
+import 'package:flutter_target/domain/repositories/user_repository.dart';
+
+class UserService implements UserRepository {
+  final options = BaseOptions(baseUrl: NetConstants.baseUrl);
+
+  Dio getDio() {
+    final dio = Dio(options);
+
+    return dio;
+  }
+
+  @override
+  void signUp(String username, String email, String gender, String password,
```

Comment:
this should be a Future<...>

Reply:
Yes! Im gonna change it when include the functionality